Applications/Hermes/hermesflaskserver.py

```diff
@@ -0,0 +1,177 @@
+    units = range(0, num)
+
+    for unit in units:
+        # TODO We don't really check for valid input here yet
+        item = "UNIT" + str(unit)
+        callsign = config.get(item, "callsign").upper()
+        nodeid = config.get(item, "nodeid")
+
+        local[str(item)] = \
+            {
+                "callsign": callsign,
+                "nodeid": nodeid,
+            }
+
+    local = json.dumps(local)
+    return json.loads(local)
+
+
+# Initialize Flask micro-framework
+app = Flask(__name__)
+
+
+@app.route('/', methods=['GET', 'POST'])
+def message():
+    """
+    Flask function that provides the "transmit" and "receive" functionality.
+
+    POST:
+
+    Transmits a provided payload data from a specified local unit to a specific remote unit.
+
+    :param localcallsign: Flask parameter - Local unit callsign to transmit from
+    :param localnodeid: Flask parameter - Local unit node ID to transmit from
+    :param destinationcallsign: Flask parameter - Remote unit callsign to transmit to
+    :param destinationnodeid: Flask parameter - Remote unit node ID to transmit to
+    :param data: Flask parameter: Data payload to transmit to remote unit
+
+    :return: Returns JSON success status for Flask server after transmission
+
+    GET:
+
+    Returns a JSON element containing the next received data from the receiver queue of the specified local unit.
+
+    :param localcallsign: Flask parameter - Local unit callsign to transmit from
+    :param localnodeid: Flask parameter - Local unit node ID to transmit from
+
+    :return: Returns JSON element containing the next received data from the receiver queue
+    """
+    # Global variables
+    global dictmsgobj
+
+    # If POST
+    if request.method == 'POST':
+        # Parse Flask arguments
+        localcallsign = request.args.get("localcallsign").upper()
+        localnodeid = request.args.get("localnodeid")
+        destinationcallsign = request.args.get("destinationcallsign").upper()
+        destinationnodeid = request.args.get("destinationnodeid")
+        data = request.args.get("data")
+
+        # Get hermes TX/RX object from global dictionary
+        unitmsgobj = dictmsgobj[localcallsign + '-' + localnodeid]
+
+        # Transmit data to remote device
+        unitmsgobj.transmit.send(destinationcallsign, int(destinationnodeid), str(data))
+
+        # Return status
+        return json.dumps(
+            {"status": "Posted Packet(s)"
+             }), 200
+
+    # If GET
+    else:
+        # Parse Flask arguments
+        localcallsign = request.args.get("localcallsign").upper()
+        localnodeid = request.args.get("localnodeid")
+
+        # Get hermes TX/RX object from global dictionary
+        unitmsgobj = dictmsgobj[localcallsign + '-' + localnodeid]
+
+        # Get next message from RX queue
+        received_item = unitmsgobj.receive.getqueueitem()
+
+        # Pickle dictionary data
+        received_data_pickle = cPickle.dumps(received_item)
```

Comment:
Don't pickle.. you already have JSON available. Much better, more secure...

Faster: https://konstantin.blog/2010/pickle-vs-json-which-is-faster/
Secure: 
![image](https://cloud.githubusercontent.com/assets/331540/24283420/e130b090-1022-11e7-8264-20c9589151c8.png)